[Registry] Rewrite according to feedback

egeloen opened the pull request:
Let's see if this version is more understable :) 

ping @php-lug/doc

tmartin commented on components/registry/index.rst:
```diff
@@ -1,12 +1,152 @@
+
+If you want to inject a registry somewhere, you should type hint against the ``ServiceRegistryInterface`` instead of
+the concrete implementation. Doing so gives you the ability to replace the default implementation by your own or by a
+third one like the lazy service registry shipped with the LugRegistryBundle.
```
did you meant "third party"?

tmartin commented:
Hell yeah! It's a lot better! :+1:

tmartin commented on components/registry/index.rst:
```diff
@@ -1,12 +1,152 @@
 
-The registry component allows you to easily manage a set of objects which all extend/implement the same class/interface.
-It is built around the ``ArrayAccess``, ``Countable`` and ``IteratorAggregate`` interfaces.
+The Lug Registry component is a very simple PHP 5.6+ implementation of the registry pattern. In a nutshell, a registry
```
Add a link to http://martinfowler.com/eaaCatalog/registry.html

egeloen commented:
@tmartin I have added the link and also added the global install doc based on the registry component.

The last part which is unclear to me is:

> IMO the only thing would be to add more explanations about each usecase, and maybe some examples of good (best?) practice.

What would you expect more?
